From 4d238d26b2bd5eab5322334d4b12de76109a0550 Mon Sep 17 00:00:00 2001
From: Olivier Roussel <olivier.roussel@inria.fr>
Date: Tue, 9 Jan 2024 10:22:34 +0100
Subject: [PATCH] Fix pybind add module segv on macos.

---
 CMake/SofaPython3Tools.cmake | 37 +-----------------------------------
 1 file changed, 1 insertion(+), 36 deletions(-)

diff --git a/CMake/SofaPython3Tools.cmake b/CMake/SofaPython3Tools.cmake
index aa9d587..7099de8 100644
--- a/CMake/SofaPython3Tools.cmake
+++ b/CMake/SofaPython3Tools.cmake
@@ -114,44 +114,9 @@ function(SP3_add_python_module)
 
     find_package(pybind11 CONFIG QUIET REQUIRED)
 
-    # We are doing manually what's usually done with pybind11_add_module(${A_TARGET} SHARED "${A_SOURCES}")
-    # since we got some problems on MacOS using recent versions of pybind11 where the SHARED argument wasn't taken
-    # into account
-    python_add_library(${A_TARGET} SHARED "${A_SOURCES}")
+    pybind11_add_module(${A_TARGET} SHARED "${A_SOURCES}")
     add_library(SofaPython3::${A_TARGET} ALIAS ${A_TARGET})
 
-    if ("${pybind11_VERSION}" VERSION_GREATER_EQUAL "2.6.0")
-        target_link_libraries(${A_TARGET} PRIVATE pybind11::headers)
-        target_link_libraries(${A_TARGET} PRIVATE pybind11::embed)
-        target_link_libraries(${A_TARGET} PRIVATE pybind11::lto)
-        if(MSVC)
-            target_link_libraries(${A_TARGET} PRIVATE pybind11::windows_extras)
-        endif()
-
-        pybind11_extension(${A_TARGET})
-        pybind11_strip(${A_TARGET})
-    else()
-        target_link_libraries(${A_TARGET} PRIVATE pybind11::module)
-
-        # Equivalent to pybind11_extension(${A_TARGET}) which doesn't exists on pybind11 versions < 5
-        set_target_properties(${A_TARGET} PROPERTIES PREFIX "" SUFFIX "${PYTHON_MODULE_EXTENSION}")
-
-        if(NOT MSVC AND NOT ${CMAKE_BUILD_TYPE} MATCHES Debug|RelWithDebInfo)
-            # Equivalent to pybind11_strip(${A_TARGET}) which doesn't exists on pybind11 versions < 5
-            # Strip unnecessary sections of the binary on Linux/macOS
-            if(CMAKE_STRIP)
-                if(APPLE)
-                    set(x_opt -x)
-                endif()
-
-                add_custom_command(
-                        TARGET ${A_TARGET}
-                        POST_BUILD
-                        COMMAND ${CMAKE_STRIP} ${x_opt} $<TARGET_FILE:${A_TARGET}>)
-            endif()
-        endif()
-    endif()
-
     set_target_properties(${A_TARGET}
         PROPERTIES
             CXX_VISIBILITY_PRESET "hidden"
-- 
2.34.1

